### PR TITLE
`Programming exercises`: Fix instructions to prepare the tests for Gradle exercises locally

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
The steps in the README for Gradle exercises that explains to instructors how they can set up the test, solution, and template repositories locally did not work.

### Description
The issues in documentation were:
- Missing `settings.gradle` file in the PLAIN_GRADLE repositories. ⇒ Gradle does not recognise the name when defining the dependency on the other template/solution module.
- Adding the dependency to the `plugins` block in `build.gradle` does not work. ⇒ It has to be added to the `dependencies` block.

Additionally, instead of having the instructor to copy-paste from the readme, I already placed the relevant changes into the `settings.gradle` and `build.gradle` but commented them out.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Create a new Java+Gradle programming exercise.
2. Clone the three `-exercise`, `-solution`, and `-tests` repositories.
3. Follow the steps in the `readme.md` of the `-tests` repository and make sure they work as intended. At the end, you should be able to run the Gradle tests in the tests repository. Gradle should report 13 passed/failed tests for the solution/template.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by yuling-sun — APPROVED** [review]
Tested on TS1. The changes work as described.

**Review by SimonPaul02 — APPROVED** [review]
Tested on TS1, works as expected! For the template, all test cases failed, for the solution, 4 were successful and 9 failed (just as you experieced it)👌

**Comment by Strohgelaender** [comment]
@SimonPaul02  That's another (not directly related) issue. The structural tests (the test that verify the test.json) seem to expect the student's classes in the assignment/src folder. With the described solution this is not the case. I personally don't think that that's a big issue though, since you can automatically generate this file on Artemis. There should be not much need to run these tests locally.

File: src/main/resources/templates/java/plain_gradle/exercise/settings.gradle [hunk 1]
```diff
@@ -0,0 +1 @@
+rootProject.name = '${exerciseNamePomXml}-Exercise'
```
[the remark this answers is not in the document]
Comment (b-fein) [reply src/main/resources/templates/java/plain_gradle/exercise/settings.gradle:1]: 1. I’d say that this file is just part of a regular Gradle project setup and should be included. It is also created when initialising a new project with `gradle init`.
2. I just copied the file from the `gradle_gradle` project type. I think there is no code in Artemis or the templates that depends on the `-Exercise` suffix. Therefore, removing it for both the Gradle and Maven project types should be fine.

[the remark this answers is not in the document]
Comment (ole-ve) [reply src/main/resources/templates/java/plain_gradle/exercise/settings.gradle]: Agree in both aspects with @b-fein.

**Comment by ole-ve** [comment]
> @SimonPaul02 That's another (not directly related) issue. The structural tests (the test that verify the test.json) seem to expect the student's classes in the assignment/src folder. With the described solution this is not the case. I personally don't think that that's a big issue though, since you can automatically generate this file on Artemis. There should be not much need to run these tests locally.

To give some context. Before executing the build plan, the source files from the template, student, or solution-repository are copied into `assignment/src` of the test repo, and then the actual build execution starts.
You could fix the behavior locally by setting the value to point to the source folder in the solution repository. E.g:
`def assignmentSrcDir = "../oletestgradle-solution/src"` (l. 23, `build.gradle` in test repository).

You could consider also adding this in some form to the README but I think it is not a big issue.

**Review by ole-ve — APPROVED** [review]
But since the issues of the actual scope were fixed, I approve.

**Comment by b-fein** [comment]
As the tests can be fixed by explaining the necessary change of just one option in the `build.gradle`, I now added it to the readme.

File: src/main/resources/templates/java/test/gradle/projectTemplate/readme.md [hunk 2]
```diff
@@ -1,37 +1,59 @@
...
+   `testImplementation(':${exerciseNamePomXml}')` or
+   `testImplementation(':${exerciseNamePomXml}-Solution')`
+   depending on which project you want to use the source files for executing the
+   test cases.
+6. Change the `assignmentSrcDir` variable in `build.gradle` to the corresponding
+   `src/` directory within the template or solution repository
+   (e.g., `../repository-solution/src/`).
+
+**Note**:
+The described changes to the `build.gradle` and `settings.gradle` files must
```
Comment (Strohgelaender) [inline src/main/resources/templates/java/test/gradle/projectTemplate/readme.md:41]: Now only the build.gradle file will get changed.
```suggestion
The described changes to the `build.gradle` files must
```

**Review by ole-ve — APPROVED** [review]
Thanks for incorporating the change, looks good to me now. 👍

**Review by Strohgelaender — APPROVED** [review]
LGTM

**Pull request merged**